### DISCUSSION TITLE
The importance of a Helix mode in Visual Studio Code.

### DISCUSSION THREAD

**Discussion by joaocarvalhoopen** [helix-editor/helix#4599] [General] [answered]
Hello,

One day, Helix will be my daily driver, but currently it isn't, like many others I use a lot of Visual Studio Code.

I know that Visual Studio Code has a VIM mode.
The movements make you much more productive.

Should You use Helix? (as a vimfluencer)
https://www.youtube.com/watch?v=5oesFDekQbs

The Primeagen rises many points, but for me all of them are irrelevant to my decision of using Helix, but there are two that are not irrelevant:

1. If you have to use Visual Studio Code (currently a standard in many organization) for something, you don't have Helix Key bindings, but you have VIM Key bindings.

2. If you have to use a remote machine, currently they only come with vi or VIM.

The second point, only time and mass adoption of Helix, will cure this one.

But the first one is really simple to correct.

It is really important for Helix key binds increased adoption that Visual Studio Code, has a Helix mode.

This will train many developers with the Helix key bindings and will make many after a while, jump ship to Helix has there daily driver.

Other things that I take from The Primeagen other videos about VIM, : vimbegood,  is that It would be really nice to have games in Helix to train people using the key bindings, like the --tutor but with iterative games.

This will be simpler with Helix future script support. 

Thank you,

Best regards,
João

**Comment by David-Else** [comment, marked as answer]
Helix keys are based on Kakoune, and there is a VS Code extension to emulate (some of) those: https://marketplace.visualstudio.com/items?itemName=reykjalin.vscode-kakoune

I doubt it would be too hard to fork that extension and expand it to be even closer to the Helix keys: https://git.sr.ht/~reykjalin/vscode-kakoune

**Reply by ghost** [reply]
there seems to be a extension to do helix stuff, https://marketplace.visualstudio.com/items?itemName=jasew.vscode-helix-emulation

**Comment by joaocarvalhoopen** [comment]
Hello David,

I didn't know that Visual Studio Code Extension existed, thank you!

But from the standpoint of Helix, there should be an extension to Visual Studio Code with the name "Helix Keys", or something like it, so that it would be easily found by the users and easily linked to Helix.

Has you said this extension can be forked and make it, to better resemble Helix key bindings. But this extension is implemented in Reason that is a kind of OCAM, at least at first view.
And that will be a barrier for many developer to contribute to the project.

Best regards,
João

**Reply by pradigunara** [reply]
Hi @joaocarvalhoopen, I stumbled upon this discussion when looking for similar things (helix mapping in vscode), and apparently someone already made usable fork of the Dance extension here: https://marketplace.visualstudio.com/items?itemName=silverquark.dancehelix

**Reply by boborbt** [reply]
I just installed it and it works great!

**Comment by nathaniel-brough** [comment]
I actually think this would be worth looking at this in a different context. I've seen [other plugins](https://github.com/vscode-neovim/vscode-neovim ) that essentially allow neovim to control the editor itself, then vscode sort of acts like a gui/frontend for the neovim instance. In my opinion this would be a great option for helix, as a vscode user you'd get the native experience of helix as well as the regular updates and bug fixes. Just food for thought.

**Reply by evelant** [reply]
idk if helix has the apis to do this but I think this would be the ideal solution. Something like vscode-neovim where neovim is actually driving the editor, but with helix. [vscode-neovim](https://github.com/vscode-neovim/vscode-neovim) would probably be a good starting place to look at how to design a solution.

**Reply by evelant** [reply]
Actually a quick look at vscode-neovim and it doesn't seem too terribly complex. If I have some time I'll try to hack on it.

**Reply by jasonwilliams** [reply]
@evelant If you start a repo please post it here

**Comment by jasonwilliams** [comment]
HI everyone

I've put together a VS Code For Helix extension which covers most of the keybindings in the Helix application. 
It's very alpha and many commands/keybindings won't work but its at a point where you can try it out

https://github.com/jasonwilliams/vscode-helix

**Reply by t-lock** [reply]
@jasonwilliams are you still working on the project? I'm using https://marketplace.visualstudio.com/items?itemName=silverquark.dancehelix which was linked above, and it works pretty well for me, but I am not sure if that project is going to reach a 1.0, or yours will?

**Reply by David-Else** [reply]
https://github.com/jasonwilliams/vscode-helix was updated last week, not sure why you think it would be discontinued. https://github.com/Silverquark/dance has not been updated for 2 years.

**Comment by samholmes** [comment]
What if we don't sleep on the second point (2); wait for helix popularity to drive `hx` availability in linux distros? What if instead we solve the problem in a different way? One idea is to somehow allow for server admin to "bring their own tools" (BYOT) to the server that they are connected to over SSH. Could it be possible to open a file over an SSH tunnel in a program that is running on the client and have reads/writes pass through the SSH connection? I imagine if there were some SSH plugin or program that can add something to the server's PATH such that any commands which the client wishes to bring were added to the this PATH dir and setting up the remote bi-directional emulation, then this would solve the "not installed here" problem we face with helix and other modern tools.

**Reply by noahfraiture** [reply]
This particular issue can be resolved with tools like sshfs
